src/cljs/main/broadfcui/page/workspace/data/utils.cljs

```diff
@@ -12,12 +13,23 @@
      {:endpoint (endpoints/get-entity workspace-id entity-type entity-name)
       :on-done (fn [{:keys [success? get-parsed-response]}]
                  (if success?
-                   (if (is-entity-set? entity-type)
-                     (let [attrs (:attributes (get-parsed-response true))
-                           items (case entity-type
-                                   "sample_set" (:items (:samples attrs))
-                                   "pair_set" (:items (:pairs attrs))
-                                   "participant_set" (:items (:participants attrs)))]
-                       (update-parent-state :selected-attr-list items :loading-attributes false))
-                     (update-parent-state :selected-attr-list (:attributes (get-parsed-response true)) :loading-attributes false))
-                   (update-parent-state :server-error (get-parsed-response false) :loading-attributes false)))})))
+                   (let [attrs (:attributes (get-parsed-response true))]
+                     (if (is-entity-set? entity-type)
+                       ;; for set entity types we display _only_ the set elements, expanded into separate rows.
+                       (let [items (case entity-type
+                                     "sample_set" (:items (:samples attrs))
+                                     "pair_set" (:items (:pairs attrs))
+                                     "participant_set" (:items (:participants attrs)))]
+                         (update-parent-state :selected-attr-list items :loading-attributes false))
```

Comment:
maybe this would be easier?
```cljs
(let [entities (case entity-type
              "sample_set" (:samples attrs))
…
  (update-parent-state :selected-attr-list (:items entities) :loading-attributes false))```